### PR TITLE
chore: exclude Node 14 and 16 on macos

### PR DESCRIPTION
**Pull request by simoneb**
Related to https://github.com/fastify/.github/issues/37.

This is just a proposal, probably not comprehensive of all the places where a similar change may have to occur, simply to capture that this is possible (and I didn't know until now) to exclude combinations, as documented in https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]
I agree with the change

Would have to be backported to `v3.y.z`

File: .github/workflows/plugins-ci.yml [hunk 1]
```diff
@@ -110,6 +110,11 @@ jobs:
       matrix:
         node-version: ${{ fromJson(inputs.node-versions) }}
         os: [macos-latest, ubuntu-latest, windows-latest]
```
Comment (climba03003) [inline .github/workflows/plugins-ci.yml:112]: Can we use `macos-latest-large`?
```suggestion
        os: [macos-latest-large, ubuntu-latest, windows-latest]
```

Comment (mcollina) [reply .github/workflows/plugins-ci.yml:112]: eventually it will migrate, as I would not count on it staying fixed. I think it's time we do this

**Review by mcollina — DISMISSED** [review]
lgtm

File: .github/workflows/plugins-ci.yml (hunk 1 above)
Comment (climba03003) [reply .github/workflows/plugins-ci.yml:112]: https://github.com/fastify/fastify/actions/runs/8874112954/job/24361249312

It doesn't seem possible to use `macos-latest-large`.

**Review by Fdawgs — COMMENTED** [review]
It needs to be applied to the rest of the workflows in this repo, does it not?

**Comment by mcollina** [comment]
yes

**Review by gurgunday — CHANGES_REQUESTED** [review]
So that someone doesn't merge it:

https://github.com/fastify/workflows/pull/127#pullrequestreview-2027678257

**Comment by mcollina** [comment]
I'm not entirely comfortable in shipping this change myself, because there is a bit of automation here that I know nothing about (I think). If somebody familiar can ship, it would unblock a few PRs that are currently failing everywhere.

**Review by gurgunday — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by gurgunday** [comment]
I’ll backport it to v3 now

**Pull request merged**